### ISSUE TITLE
Pinebook Pro NVMe NixOS

### ISSUE THREAD

**Issue by serpent213** [Tow-Boot/Tow-Boot#263]
I really like [the idea of Tow-Boot](https://discourse.nixos.org/t/planning-for-a-better-nixos-on-arm-and-other-non-x86-64-systems), great piece of software!

In my recent PBP I installed a Crucial P3 1TB NVMe and now I'm wondering whether it's possible to boot from the new drive and leave Manjaro alone for now on the eMMC. My main issue is the kernel apparently not detecting the drive on boot.

![IMG_2912](https://github.com/Tow-Boot/Tow-Boot/assets/106662/c8a913bc-570a-4546-a792-8ec0c539de95)

https://github.com/NixOS/nixpkgs/issues/219330 has some more screenshots.

NixOS Channel 23.05 with Linux 6.4.12, Tow-Boot 2022.07-006 and *development* (easy to build on Manjaro, I like the build process!  🙂).

Also I noticed that the keyboard (ANSI with [latest firmware](https://github.com/dragan-simic/pinebook-pro-keyboard-updater)) regularly loses a stroke, like 1 in 5 or so. Further, does not work to start kernel shell on fail. External USB keyboard, no joy, either.

**Comment by tomfitzhenry** [comment]
Yes, booting directly from NVMe works. I do this on NixOS: https://github.com/NixOS/nixpkgs/issues/219330#issuecomment-1717366723

**Comment by blmayer** [comment]
> Also I noticed that the keyboard (ANSI with [latest firmware](https://github.com/dragan-simic/pinebook-pro-keyboard-updater)) regularly loses a stroke, like 1 in 5 or so. Further, does not work to start kernel shell on fail. External USB keyboard, no joy, either.

Interesting, in my case I have a duplicate key, didn't notice any regularity, since it is not 1 in 5, it seems to be 1 in 20 or so.

I'mm using Archlinux ARM.

**Comment by tomfitzhenry** [comment]
https://github.com/NixOS/nixos-hardware/pull/743 is my nixos-hardware PR adding NVMe support for Pinebook Pro on NixOS.
